File: src/main/java/com/github/hhhzzzsss/songplayer/commands/SkipCommand.java
package com.github.hhhzzzsss.songplayer.commands;

import com.github.hhhzzzsss.songplayer.playing.SongPlayer;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;

class SkipCommand extends Command {
    @Override
    public String getName() {
        return "skip";
    }

    @Override
    public String getDescription() {
        return "Skips current song";
    }

    @Override
    public void buildNode(LiteralArgumentBuilder<FabricClientCommandSource> node) {
        node.executes(context -> {
            if (SongPlayer.instance.currentSong == null) {
                com.github.hhhzzzsss.songplayer.SongPlayer.addChatMessage("§6No song is currently playing");
                return 1;
            }

            com.github.hhhzzzsss.songplayer.SongPlayer.addChatMessage("§6Skipped the current song.");
            SongPlayer.instance.currentSong = null;

            return 1;
        });
    }
}

File: src/main/java/com/github/hhhzzzsss/songplayer/commands/SkipCommand.kt
package com.github.hhhzzzsss.songplayer.commands

import com.github.hhhzzzsss.songplayer.SongPlayer
import com.github.hhhzzzsss.songplayer.playing.SongHandler
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource

internal class SkipCommand : Command {
    override val name = "skip"

    override val description = "Skips current song"

    override fun buildNode(node: LiteralArgumentBuilder<FabricClientCommandSource>) {
        node.executes {
            if (SongHandler.instance.songQueue.isEmpty) {
                SongPlayer.addChatMessage("§6No song is currently playing")
                return@executes 1
            }
            SongPlayer.addChatMessage("§6Skipped the current song.")
            SongHandler.instance.songQueue.next()
            1
        }
    }
}